### PR TITLE
MSC2002: Proposal for adopting MSC1884 as v4 rooms

### PR DESCRIPTION
**Pull request by ara4n**
Rendered: https://github.com/matrix-org/matrix-doc/blob/matthew/room_v4/proposals/2002-rooms-v4.md

### CODE REVIEW

File: proposals/2002-rooms-v4.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# MSC 2002 - Rooms V4
+
+This MSC proposes creating a new room version named v4 to allow servers to switch
+room ID grammar to that proposed in MSC1884.
+
+## Proposal
+
+The new room version is called "4". The only difference between v4 and v3 is
+that v4 rooms use the grammar for defining room IDs as proposed in MSC1884 -
```
Comment (turt2live) [inline proposals/2002-rooms-v4.md]:
```suggestion
that v4 rooms use the grammar for defining event IDs as proposed in MSC1884 -
```

and throughout

Comment (ara4n) [reply proposals/2002-rooms-v4.md]: oops.

File: proposals/2002-rooms-v4.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# MSC 2002 - Rooms V4
+
+This MSC proposes creating a new room version named v4 to allow servers to switch
+room ID grammar to that proposed in MSC1884.
```
Comment (Cadair) [inline proposals/2002-rooms-v4.md]: event ID?

Comment (ara4n) [reply proposals/2002-rooms-v4.md]: oops, thinko, thanks, fixed

**Comment by ara4n** [comment]
@mscbot fcp merge

we want to get this and 1884 merged asap and released in prep for 1.0

**Comment by anoadragon453** [comment]
#1884 - link for the lazy

**Comment by richvdh** [comment]
I think what happens for room v5 / matrix 1.0 and beyond are somewhat orthogonal here. +1 to getting features out into the wild rather than sitting on them until they form a huge batch, so I am happy with this.

**Comment by ara4n** [comment]
> I think what happens for room v5 / matrix 1.0 and beyond are somewhat orthogonal here.

I think the proposed longer-term context is fairly important to try to understand how this fits in, but agreed it's technically orthogonal to the MSC itself.

**Comment by ara4n** [comment]
@matrix-org/core-team I'd like to propose an expedited FCP for this, so we can merge this (and #1884) into the spec and ship it in synapse (https://github.com/matrix-org/synapse/pull/5210) asap, so that when 1.0 rolls around we have as many servers talking v4 rooms in the wild as possible and we can enable creating v4 rooms by default.

I suggest we reduce FCP to 1 day - i.e. aim to merge this around 15:00 GMT+1 tomorrow, unless someone finds a last minute problem with MSC1884 (or this).

**Comment by richvdh** [comment]
Impl is at https://github.com/matrix-org/synapse/pull/5217, ready to merge when this completes FCP.

**Comment by richvdh** [comment]
as per https://github.com/matrix-org/matrix-doc/pull/2002#issuecomment-494023657, we are declaring FCP to be complete here.

**Comment by richvdh** [comment]
cancelling the bot's FCP to stop it "helping"

@mscbot fcp cancel

**Comment by turt2live** [comment]
Spec PR: https://github.com/matrix-org/matrix-doc/pull/2019

**Comment by turt2live** [comment]
merged 🎉

**Pull request merged**